src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -777,6 +792,11 @@ private void EnforceDoesNotReturn(SyntaxNode? syntaxOpt)
             {
                 DebugVerifier.Verify(analyzedNullabilitiesMap, snapshotManager, node);
             }
+
+            if (avoidRewrite)
+            {
+                return node;
```

Comment:
>return node [](start = 16, length = 11)

Would it make sense to push this down to the Rewrite to disable actual node modifications, but do all the other work?

Reply:
Responded in the other comment, but I don't believe that running the rewriter actually gets us any real verification and it would significantly complicate the generated code.

---
In reply to: [402669321](https://github.com/dotnet/roslyn/pull/42403#discussion_r402669321) [](ancestors = 402669321)